lib/common/strings.c

```diff
@@ -952,12 +933,25 @@ str_any_of(bool casei, const char *s, va_list args)
 bool
 pcmk__str_any_of(const char *s, uint32_t flags, ...)
 {
+    GList *lst = NULL;
     va_list ap;
     bool rc;
 
     va_start(ap, flags);
-    rc = str_any_of(pcmk_is_set(flags, pcmk__str_casei), s, ap);
+    while (1) {
+        const char *ele = va_arg(ap, const char *);
+
+        if (ele == NULL) {
+            break;
+        } else {
+            lst = g_list_prepend(lst, (gpointer) ele);
+        }
+    }
     va_end(ap);
+
+    rc = pcmk__str_in_list(lst, s, flags);
+
+    g_list_free(lst);
```

Comment:
Not a fan of this, the glib overhead (especially memory handling) will be significant in loops

Reply:
With pcmk__str_star_matches support in pcmk__strcmp, pcmk__str_any_of can just be a loop over the varargs as well and then we don't need to do any list construction here.